### DISCUSSION TITLE
How can I use `goVersion.binaryPath` for installing tools.

### DISCUSSION THREAD

**Discussion by rezigned** [golang/vscode-go#1691] [Help]
In the following code. `getCurrentGoRoot()` always return some value. So, it always evaluates to first ternary value (i.e. `path.join(...)`. 

My question, how can I force it to use `goVersion.binaryPath` instead? Setting `GOROOT=` will do that but it'll break lot of things. Thank you.

https://github.com/golang/vscode-go/blob/d1d61bc5e6f1d577e11a893eaeee49e88933e461/src/goInstallTools.ts#L239-L241

**Comment by stamblerre** [comment]
One option to switch your Go installation is to set:

```json5
"go.alternateTools": {
    "go": "path/to/go",
}
```

One other option is to use the Go extension's built in options for managing your Go installation. On the bottom left of the VS Code window, you should see a button with your Go version listed. You can click on that button and select "Choose Go environment" to switch your Go version.

**Reply by rezigned** [reply]
Thanks @stamblerre, I've tried your options. But for some reason it doesn't work for tools installation (It works for "Install current package" though). 

My guess is that. When the extension loads, it'll always set `setCurrentGoRoot` to whatever output from go runtime (e.g. from `PATH` env) `go env ...`.
So, `getCurrentGoRoot()` always returns that path for tools installation.

https://github.com/golang/vscode-go/blob/d1d61bc5e6f1d577e11a893eaeee49e88933e461/src/goInstallTools.ts#L494-L497

But for `Install Current Package`. It uses `getBinPath("go")` which works fine.

https://github.com/golang/vscode-go/blob/33339e687f2034f80ae3018a79db121c8e04feed/src/goInstall.ts#L29

Why we can't use `getBinPath("go")` for installing tools also (as shown in the question).

**Comment by stamblerre** [comment]
Ah sorry didn't see that you wanted to change the tools installation path. Does `go.toolsGopath` work for you?

**Reply by rezigned** [reply]
Thanks for your help! Maybe my question is not very clear. I'm not trying to change Go's tool paths. 

I want `Go Install/Update Tools` command to use `go` from runtime

https://github.com/golang/vscode-go/blob/33339e687f2034f80ae3018a79db121c8e04feed/src/goInstall.ts#L29

But not from 

https://github.com/golang/vscode-go/blob/d1d61bc5e6f1d577e11a893eaeee49e88933e461/src/goInstallTools.ts#L239-L241

Not sure if it's possible.
